### PR TITLE
Add MT103 to Json Transformer

### PR DESCRIPTION
**Pull request by sineththamuditha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT103ToJsonTransformer.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+        return mt103Message.toString();
+    }
+
+    /**
+     * Method to replace amount strings as numbers from MT103 model JSON.
+     *
+     * @param textBlock JSON conversion of the model {@link MT103Message} (converted using Gson).
+     */
+    private static void replaceAmountAsNumber(JsonObject textBlock) {
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/MT103ToJsonTransformer.java:111]: Shall we move these common methods to common util class as discussed during the code review

File: src/main/java/org/wso2/carbon/module/swiftiso20022/validation/common/ValidationResult.java [hunk 2]
```diff
@@ -30,6 +30,10 @@ public ValidationResult() {
         this(true, null, null);
     }
 
+    public ValidationResult(boolean isValid) {
+        this(isValid, null, null);
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/validation/common/ValidationResult.java:34]: Why do we need this constructor

Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/validation/common/ValidationResult.java]: It was added to improve readability.
Removed the boolean constructor in commit fdf9fe08dfb8b8c04ee04cb393c773ffe3ab319b and replaced used places with the default constructor with comments for clarification.

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/format/validators/MTBlockFormatValidator.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+public class MTBlockFormatValidator {
+
+    private static final Log log = LogFactory.getLog(MTBlockFormatValidator.class);
+
+    /**
+     * This regex captures global match in the format -> one or more of {tag:value} .
+     * tag -> 3 digits
+     * value -> one or more character except {,},: and line break
+     */
+    private static final String USER_HEADER_BLOCK_FORMAT_REGEX = "^(\\{\\d{3}:[^}]{1,36}})+$";
```
Comment (lalaji) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/format/validators/MTBlockFormatValidator.java:47]: some of the regex patterns are defined in a separate class while some are as class  variables. What's the reason?

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field20.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+        if (!OPTIONS.contains(option)) {
+            throw new MTMessageParsingException(String.format(MTParserConstants.INVALID_OPTION_FOR_FIELD, option,
+                    ConnectorConstants.FIELD_20));
+        }
+
+        Matcher field20Matcher = MTParserConstants.FIELD_20_REGEX_PATTERN.matcher(field20String);
+
+        if (field20Matcher.matches()) {
+
```
Comment (lalaji) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field20.java:89]: is this new line needed?

**Review by lalaji — APPROVED** [review]

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field23.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        Matcher field23Matcher = MTParserConstants.FIELD_23_REGEX_PATTERN.matcher(field23String);
+
+        if (field23Matcher.matches()) {
+
+            // group 1 -> Code
+            // group 2 -> /Additional Information
+            // group 3 -> Additional Information
+            return new Field23(option, field23Matcher.group(1), field23Matcher.group(3));
+
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field23.java:130]: shall we remove these extra lines

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field32.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        if (!OPTIONS.contains(option)) {
+            throw new MTMessageParsingException(String.format(MTParserConstants.INVALID_OPTION_FOR_FIELD, option,
+                    ConnectorConstants.FIELD_32));
+        }
+
+        // Get matcher to the regex matching -> (Date)(Currency)(Amount)
+        Matcher field32Matcher = MTParserConstants.FIELD_32_REGEX_PATTERN.matcher(field32String);
+
+        if (field32Matcher.matches()) {
+
```
Comment (Ashi1993) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field32.java:123]: Lets remove these extra lines. Check all places

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/format/validators/MTBlockFormatValidator.java (hunk 3 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/format/validators/MTBlockFormatValidator.java:47]: These regexes are specifc regexes which can only be used for format validation. So there is no need for them to be present publicly.

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field20.java (hunk 4 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field20.java]: Removed in commit 7b56b383e3127da22c1b7305f9f84c495346f0f3.

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field23.java (hunk 5 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field23.java]: Remove in commit 7b56b383e3127da22c1b7305f9f84c495346f0f3.

File: src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field32.java (hunk 6 above)
Comment (sineththamuditha) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/mt/models/fields/Field32.java]: Removed all extra lines from field models in commit 7b56b383e3127da22c1b7305f9f84c495346f0f3.

**Review by lalaji — APPROVED** [review]

**Review by Ashi1993 — APPROVED** [review]

**Pull request merged**